### PR TITLE
Sign msi engine

### PR DESCRIPTION
**Pull request by chcosta**
Fixes https://github.com/dotnet/core-eng/issues/10771

Extract, sign, and reattach msi engine's

### CODE REVIEW

**Comment by chcosta** [comment]
Validated locally, internal build validation is in progress.

File: src/Microsoft.DotNet.SignTool.Tests/SignToolTests.cs [hunk 1]
```diff
@@ -727,6 +727,52 @@ public void ZipFile()
             });
         }
 
+        [SkippableFact]
```
Comment (jcagme) [inline src/Microsoft.DotNet.SignTool.Tests/SignToolTests.cs:730]: In my PR there are PostBuild test cases. We'll need to add a test case for this depending on which PR goes in first

Comment (jcagme) [reply src/Microsoft.DotNet.SignTool.Tests/SignToolTests.cs:731]: My previous comment won't be valid anymore once I work on feedback

File: src/Microsoft.DotNet.SignTool/src/BatchSignUtil.cs [hunk 2]
```diff
@@ -120,6 +120,57 @@ bool signFiles(IEnumerable<FileSignInfo> files)
...
+                                                fileInfo.IsWixContainer() &&
+                                                Path.GetExtension(fileInfo.FullPath) == ".exe").ToArray();
+
+                if (enginesToSign.Length == 0)
+                {
+                    return true;
+                }
+
+                Dictionary<string, FileSignInfo> engines = new Dictionary<string, FileSignInfo>();
+                var workingDirectory = $"{Path.Combine(_signTool.TempDir, "engines")}";
```
Comment (jcagme) [inline src/Microsoft.DotNet.SignTool/src/BatchSignUtil.cs:135]: I think you can just do `Path.Combine(_signTool.TempDir, "engines")` without the $""

File: src/Microsoft.DotNet.SignTool/src/BatchSignUtil.cs [hunk 3]
```diff
@@ -120,6 +120,57 @@ bool signFiles(IEnumerable<FileSignInfo> files)
...
+                foreach (var engine in engines)
+                {
+                    _log.LogMessage(MessageImportance.Normal, $"Attaching engine {engine.Key} to {engine.Value.FullPath}");
+                    int exitCode = RunWixTool("insignia.exe", $"-ab {engine.Key} {engine.Value.FullPath} -o {engine.Value.FullPath}", workingDirectory, _signTool.WixToolsPath);
+                    if (exitCode != 0)
+                    {
+                        _log.LogError($"Failed to attach engine to {engine.Value.FullPath}");
+                        return false;
+                    }
+
```
Comment (jcagme) [inline src/Microsoft.DotNet.SignTool/src/BatchSignUtil.cs:169]: Remove blank

**Review by jcagme — APPROVED** [review]
Few nits

**Review by mmitche — APPROVED** [review]

**Pull request merged**